### DISCUSSION TITLE
How to use connectOrCreate with many to many

### DISCUSSION THREAD

**Discussion by jsun969** [prisma/prisma#8851] [Q&A] [answered]
I want to use Prisma to associate articles with tags, that is: A post has multiple tags and a tag belongs to multiple posts  
But the example in Prisma will result in the creation of duplicate tags

Here is my Prisma model
```prisma
model Article {
  id        Int      @id @default(autoincrement())
  createdAt DateTime @default(now())
  title     String
  summary   String
  link      String   @unique
  archive   String
  content   String
  image     String
  tags      Tag[]
}

model Tag {
  id       Int       @id @default(autoincrement())
  name     String
  articles Article[]
}
```
This is my code according to the documentation, it causes duplicate tags to appear
```js
await prisma.article.create({
  data: {
    title,
    summary,
    link,
    archive,
    content,
    image,
    tags: {
      create: tags.map((tag) => ({ name: tag })),
    },
  },
});

```
When I use connectOrCreate, it reports an error in many-to-many mode
```js
await prisma.article.create({
  data: {
    title,
    summary,
    link,
    archive,
    content,
    image,
    tags: {
      connectOrCreate: {
        where: tags.map((tag) => ({ name: tag })),
        create: tags.map((tag) => ({ name: tag })), 
      },
    },
  },
});
```

**Comment by ryands17** [comment, marked as answer]
@jsun969 👋 

There's an error with where you're using `tags.map`. You need to map over `connectOrCreate` like this:

```ts
await prisma.article.create({
  data: {
    title,
    summary,
    link,
    archive,
    content,
    image,
    tags: {
        connectOrCreate: tags.map(tag => ({
          where: { name: tag }, create:  { name: tag } 
        }))
    },
  },
});
```

**Reply by jsun969** [reply]
Thanks, I used your code after setting the name to unique, and it worked!

**Reply by sawirricardo** [reply]
I think there should be a way to not make the name field unique.

**Reply by DonKoko** [reply]
Agreed. In our case we want to do connectOrCreate based on name, even if its not unique. Considering this we cant really do it so we have to make it the slow way, querying and checking if the entry exists and performing an action based on that.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
